src/cmd/services/m3aggregator/config/aggregator.go

```diff
@@ -846,3 +867,37 @@ func setMetricPrefix(
 	}
 	return fn([]byte(*str))
 }
+
+func (c *AggregatorConfiguration) newPassThroughWriter(
+	cs client.Client,
+	iOpts instrument.Options,
+	shardFn sharding.AggregatedShardFn,
+) (writer.Writer, error) {
+	// temporary change to use a separate m3msg topic for pass-through metrics during the migration.
+	flushCfg := (*c).Flush // making a copy to avoid mutating original.
+	for _, handler := range flushCfg.Handlers {
+		if handler.DynamicBackend != nil && c.PassThroughTopicName != nil {
+			handler.DynamicBackend.Producer.Writer.TopicName = *c.PassThroughTopicName
+		}
+	}
```

Comment:
Hmm... not the most friendly code, esp for how simple it is. IMO further datapoints to not use config pointers if we can avoid it.

Reply:
talked offline: going to leave as is because the changes are pretty pervasive to make handler not be a pointer type.